packages/admin-ui/src/switch/switch.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+import type { Checkbox as ReakitCheckbox } from 'reakit/Checkbox'
+import { createComponent } from '@vtex/admin-ui-react'
+
+import { useCheckboxState } from '../components/Checkbox'
+import { Inline } from '../inline'
+import { SwitchButton } from './switch-button'
+import { Stack } from '../stack'
+import { Label } from '../components/Label'
+import { SwitchMessage } from './message'
+import type { SwitchOptions } from './types'
+import { Text } from '../components/Text'
+
+export const Switch = createComponent<typeof ReakitCheckbox, SwitchOptions>(
+  (props) => {
+    const { state, helpText, errorText, label, error, disabled, ...htmlProps } =
+      props
+
+    const hasError = !!(error && errorText)
+    const hasMessage = !!(hasError || helpText)
+
+    return (
+      <Inline hSpace="$m" vSpace="">
+        <SwitchButton state={state} disabled={disabled} {...htmlProps} />
+        <Stack space="$xs">
+          {label && (
+            <Text as={Label} tone={disabled ? 'secondary' : 'primary'}>
+              {label}
+            </Text>
+          )}
+          {hasMessage && (
+            <>
+              {helpText && <SwitchMessage helpText={helpText} />}
```

Comment:
The way you create the Message component you just need to pass errorText and helpText to the component and both messages will appear:
```suggestion
             <SwitchMessage helpText={helpText} errorText={errorText} />
```

Reply:
Truestory! Good catch!